grade-scripts/shared/utils.js

```
@@ -13,6 +15,53 @@ const responseIsOK = (response, siteToCheck) => {
         firstStatusDigit === 3
 }
 
+const getSiteData = (inputPath, fileForSubset) => {
+    // get initial file data
+    let siteDataFiles = fs.readdirSync(inputPath)
+
+    // if we've defined a file for subset, get all the sites listed in that
+    // and avoid processing all the rest
+    if (fileForSubset) {
+        let sitesForSubset = fs.readFileSync(fileForSubset, { encoding: 'utf8' })
```

Can we also fail here if the file is invalid?